CMake 4 won't build charm++ v7.0.0

The issue is line 13 of charm++'s CMakeLists.txt, which says
```
cmake_minimum_required(VERSION 3.4)
```
This makes CMake complain:
```
CMake Error at CMakeLists.txt:13 (cmake_minimum_required):
Compatibility with CMake < 3.5 has been removed from CMake.

Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
to tell CMake that the project requires at least <min> but has been updated
to work with policies introduced by <max> or earlier.
```
CMake 4 is now the default that's installed on macos, so tests fail.

Technically this is an upstream issue, and charm++ has already changed it in charmplusplus/charm#3882 . But for v7.0.0, I found that simply changing it to `cmake_minimum_required(VERSION 3.5)` allows charm to build — they must not be using any CMake features that were deprecated in the change from 3.4 to 3.5.

--- a/.github/workflows/Tests.yaml
+++ b/.github/workflows/Tests.yaml
@@ -1022,6 +1022,8 @@ ${{ matrix.build_type }}-pch-${{ matrix.use_pch || 'ON' }}"
           spack env create spectre
           spack env activate spectre
           spack add blaze@3.8.2 ~blas ~lapack smp=none
+          # Support installing charm 7.0.0 with CMake 4.0
+          export CMAKE_POLICY_VERSION_MINIMUM=3.5
           spack add charmpp@7.0.0 +shared backend=multicore build-target=charm++
           # Use main branch until spack has 2.0 release
           spack add libxsmm@main
@@ -1110,6 +1112,7 @@ ${{ matrix.build_type }}-pch-${{ matrix.use_pch || 'ON' }}"
             -D CMAKE_INSTALL_PREFIX=../install \
             -D BUILD_DOCS=OFF \
             -D USE_XSIMD=OFF \
+            -D CMAKE_OSX_SYSROOT=macosx \
             $GITHUB_WORKSPACE
       - name: Build unit tests
         working-directory: build

